actors/builtin/reward/reward_test.go

```diff
@@ -1 +1,191 @@
 package reward_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/filecoin-project/specs-actors/actors/abi"
+	"github.com/filecoin-project/specs-actors/actors/abi/big"
+	"github.com/filecoin-project/specs-actors/actors/builtin"
+	"github.com/filecoin-project/specs-actors/actors/builtin/reward"
+	"github.com/filecoin-project/specs-actors/actors/runtime/exitcode"
+	"github.com/filecoin-project/specs-actors/actors/util/adt"
+	"github.com/filecoin-project/specs-actors/support/mock"
+	tutil "github.com/filecoin-project/specs-actors/support/testing"
+)
+
+func TestConstructor(t *testing.T) {
+	actor := rewardHarness{reward.Actor{}, t}
+
+	receiver := tutil.NewIDAddr(t, 1000)
+	builder := mock.NewBuilder(context.Background(), receiver).WithCaller(builtin.SystemActorAddr, builtin.SystemActorCodeID)
+	rt := builder.Build(t)
+	actor.constructAndVerify(rt)
+}
+
+func TestAwardBlockReward(t *testing.T) {
+	actor := rewardHarness{reward.Actor{}, t}
+
+	receiver := tutil.NewIDAddr(t, 1000)
+	builder := mock.NewBuilder(context.Background(), receiver).WithCaller(builtin.SystemActorAddr, builtin.SystemActorCodeID)
+
+	t.Run("block reward with no payable reward", func(t *testing.T) {
```

Comment:
More specific "no reward when penalty equals reward and reward actor has no balance except gas".

We should also check for penalty exceeding reward.